packages/@lwc/template-compiler/src/shared/types.ts

```diff
@@ -99,7 +99,7 @@ export interface IRElement {
 
 export interface IRText {
     type: 'text';
-    value: string | TemplateExpression;
+    value: Array<string | TemplateExpression>;
```

Comment:
addressed as part of: https://github.com/salesforce/lwc/pull/2405/commits/c9dae76067b207e826078f7c679a328294623b71

Reply:
I was thinking about creating multiple AST nodes instead of a single one for interpolated expressions. For example:

```
Hello {name}!
```

Produces:

```js
[
  { type: 'text', value: 'Hello ' },
  { type: 'textInterpolation', value: { type: 'Identifier', name: 'name' } },
  { type: 'text', value: '!' },
]
```

Let merge it as is and refactor this with the template compiler AST refactor planned for next release.

Reply:
I have been thinking about this over the weekend. I think we should revert to the previous AST shape before merging by keeping independent AST nodes for each part of the text interpolation.